src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -161,6 +204,161 @@ static int gavl_intersect_cmp_func(uintptr_t ustart, uintptr_t len, gavl_tree_no
     return cmp_ret;
 }
 
+/* gavl_tree_node_merge merges the subnode_list of src node into dest node and
+ * update dest node starting addr and len. This function is called during insert
+ * when we find inserted node intersects existing tree node and needs to be merged. */
+static int gavl_tree_node_merge(const gavl_tree_node_s * src, gavl_tree_node_s * dest)
+{
+    gavl_tree_subnode_s *iter;
+    uintptr_t src_end = src->addr + src->len;
+    uintptr_t dest_end = dest->addr + dest->len;
+
+    if (src->subnode_list == NULL || src_end <= dest->addr || dest_end <= src->addr)
+        goto fn_exit;
+
```

Comment:
> Is there any case where the above conditions can be true? 

In my current code, there is no such case. But, I don't want to put an assumption that subnode_list must be non-empty in the function. 
This function should be an independent function and should not know what kind of `src` and `dest` the user will give. Doing a check will not hurt anything, on the contrary, it makes this function safer and reduces bugs and code errors.

Reply:
This is only an internal routine which is never exposed to external. Your concern seems too conservative. As the downside of these checks, they add an additional branch and several condition checks.  The instructions overhead is not a big problem since this function is called only in `insert`, but as I said, they are unnecessary.

Reply:
alright, I can delete it